acvm_backend_barretenberg/src/composer.rs

```diff
@@ -101,30 +120,37 @@ impl StandardComposer {
         result.to_vec()
     }
 
-    pub fn create_proof_with_pk(
-        &mut self,
+    fn create_proof_with_pk(
+        &self,
+        constraint_system: &ConstraintSystem,
         witness: WitnessAssignments,
         proving_key: &[u8],
     ) -> Vec<u8> {
-        let cs_buf = self.constraint_system.to_bytes();
+        let circuit_size = self.get_circuit_size(constraint_system);
+        let CRS {
+            g1_data, g2_data, ..
+        } = CRS::new(circuit_size as usize);
```

Comment:
Does this actually use g1_data? Because it might be more efficient to use a `G2::new` instead